update trigger function to support environment variables

osos opened the pull request:
update trigger function to support environment variables for both wait and loop functionality

<!--
## Before you hit that Submit button....

Please read our [contribution guidelines](https://github.com/node-red/node-red/blob/master/CONTRIBUTING.md)
before submitting a pull-request.

## Types of changes

What types of changes does your code introduce?
Put an `x` in the boxes that apply
-->

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

*unsure* if breaking, but I dont think so. I am just a novice user.

<!--
If you want to raise a pull-request with a new feature, or a refactoring
of existing code, it **may well get rejected** if it hasn't been discussed on
the [forum](https://discourse.nodered.org) or
[slack team](https://nodered.org/slack) first.

-->

## Proposed changes

<!-- Describe the nature of this change. What problem does it address? -->
This fix introduces a variable "triggerfunction" to hold the chosen function of the trigger node.

This allows the duration variable to solely by used for exactly that the 'duration'. This also allows for the use of environment variables as the duration, for both wait and loop functionality.

Fixes #2528 

## Checklist
<!-- Put an `x` in the boxes that apply -->

- [ ] I have read the [contribution guidelines](https://github.com/node-red/node-red/blob/master/CONTRIBUTING.md)
- [ ] For non-bugfix PRs, I have discussed this change on the forum/slack team.
- [ ] I have run `grunt` to verify the unit tests pass
- [ ] I have added suitable unit tests to cover the new/changed functionality

dceejay commented:
great - thanks - will have a look over the weekend.

dceejay commented:
Looks good to me - works fine on my old flows, and passes tests.
@knolleary  - Do you want this in master or dev ?

knolleary commented:
As long as we are 100% certain the the logic properly migrates existing node configurations to the new scheme? The use of negative duration to indicate one mode over another was always a horrible hack 😉 and I'll be glad to see the back of it.

Two requests:

 - rename `triggerfunction` to `func` - we do try to keep node property names short, within reason.
 - add some new unit tests to https://github.com/node-red/node-red/blob/master/test/nodes/core/function/89-trigger_spec.js

osos commented:
renamed "triggerfunction" to "func".
also added some additional lines to ensure backward-compatibility.

Not familiar with unit test, so haven't touched those

knolleary commented:
Thanks - I guess we can look at adding the tests.

This will have to go into the `dev` branch for the 1.1.0 release as it changes the node configuration model.

EasyPP commented:
Had hoped for this in 1.1.

When do plan to merge this one?
